CMSgov/mdct-mcr
# MDCT-471: ChoiceList component does not correctly update form state/field value onChange

## Description
Fixes [MDCT-471](https://qmacbis.atlassian.net/browse/MDCT-471). The Choicelist component is updated to keep track of its individual choices and return them

### How to test

Before testing, add:

```
{
      id: "abf-radio",
      type: "radioList",
      required: true,
      props: {
        name: "Choice List Radio Name",
        type: "radio",
        label: "Choice List Radio Label",
        choices: [
          { label: "Choice 1", value: "A" },
          { label: "Choice 2", value: "B" },
          { label: "Disabled choice 3", value: "C" },
        ],
      },
      validation: {
        type: "array",
        options: { min: 1 },
        errorMessages: {
          min: "Description text is required",
        },
      },
    },
    {
      id: "abf-check",
      type: "checkboxList",
      props: {
        name: "Choice List Checkbox Name",
        type: "checkbox",
        label: "Choice List Checkbox Label",
        choices: [
          { label: "Choice 1", value: "A" },
          { label: "Choice 2", value: "B" },
          { label: "Disabled choice 3", value: "C" },
        ],
      },
      validation: {
        type: "array",
        options: { min: 1 },
        errorMessages: {
          min: "Description text is required",
        },
      },
    },
```

to adminBannerForm.ts. You'll want to add it as part of the "Fields" array

- ./dev local
- Sign in as an admin
- My Account -> Manage Account -> Banner Editor
- Play with the checkboxes and console log form.getValues() in the useEffect hook

### Changed Dependencies


## Code author checklist

## Reviewer checklist (two different people)
